packages/codemods/src/codemods/v2.2.x/updateClerk/updateClerk.ts

```diff
@@ -0,0 +1,61 @@
+import type { FileInfo, API } from 'jscodeshift'
+
+export default function transform(file: FileInfo, api: API) {
+  const j = api.jscodeshift
+
+  const ast = j(file.source)
+
+  ast
+    .find(j.ImportDeclaration, { source: { value: '@clerk/clerk-react' } })
+    .forEach((importDeclaration) => {
+      importDeclaration?.value.specifiers?.forEach((specifier) => {
+        if (
+          j.ImportSpecifier.check(specifier) &&
+          specifier.imported.name === 'withClerk'
+        ) {
+          // Found `withClerk` import. Now I want to replace that with a
+          // `ClerkLoaded` import instead
+          specifier.imported.name = 'ClerkLoaded'
+
+          // And finally, for the imports, we need to add a new import to give us `navigate`
+          j(importDeclaration).insertAfter(
+            "import { navigate } from '@redwoodjs/router'"
+          )
+        }
+      })
+    })
+
+  // Remove old RW Clerk components
+  ast.find(j.VariableDeclaration).forEach((variableDeclaration) => {
+    if (
+      variableDeclaration.value.declarations.find((declaration) => {
+        return (
+          j.VariableDeclarator.check(declaration) &&
+          j.Identifier.check(declaration.id) &&
+          (declaration.id.name === 'ClerkAuthProvider' ||
+            declaration.id.name === 'ClerkAuthConsumer')
+        )
+      })
+    ) {
+      j(variableDeclaration).remove()
```

Comment:
This removes all the comments above too; I'll try to preserve them.

Reply:
@Tobbe to keep comments you could do something like this:

```js
      // assuming `comments` is defined earlier

      if (variableDeclaration.value.comments) {
        comments.concat(
          variableDeclaration.value.comments.map((comment) => comment.value)
        )
      }
```

Reply:
And then you'd have to insert them back later

Reply:
What a pain it was to insert them back in! I wanted to put them before the new `ClerkAuthProvider`. But for some reason, when I inserted it using `insertBefore('const ClerkAuthProvider = () => ....')` I couldn't find it with `ast.find(j.VariableDeclaration,`. So I had to construct a separate, proper, `VariableDeclaration`. Attach the comments to that. And finally insert that whole thing into the ast.